Update agency company information forms

oikeme opened the pull request:
Update the error message when on the agency sign up form and company update form to indicate when an invalid domain name has been entered.

#### Proposed Changes

* Update the error messaging for creating and updating company information to include when domain information is entered but invalid.

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Apply this PR.
* Run `yarn-start-jetpack-cloud` and visit `http://jetpack.cloud.localhost:3000/partner-portal/company-details`.
* Attempt to save the `Company website` field without a value and ensure that this still returns an error indicating that the value is required.
* Attempt to save the same field with an invalid domain name (a space in the domain, or no TLD) and ensure that the error message also indicates that the value must be valid.

<img width="641" alt="Screenshot 2022-12-12 at 4 26 00 PM" src="https://user-images.githubusercontent.com/1273880/207181322-e5fc9163-30bb-409d-b872-e577f9f8b6c3.png">


#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

oikeme committed:
Update agency company information forms

Update the error message when on the agency sign up form and company update form to indicate when an invalid domain name has been entered.

yashwin commented on client/state/partner-portal/partner/utils.ts:
```diff
@@ -23,6 +23,13 @@ export function translateInvalidPartnerParameterError( parameters: object ) {
 		return labels[ field ] || field;
 	} );
 
+	if ( 'company_website' in details ) {
```
I'm a bit concerned about the `details` being undefined here. Will `details` always have some value(not undefined)?

yashwin commented on client/state/partner-portal/partner/utils.ts:
```diff
@@ -6,7 +6,7 @@ import { sprintf, __ } from '@wordpress/i18n';
  * @param {object} parameters Parameters with errors.
```
Can we also include `details` here as param?

yashwin reviewed (changes requested):
LGTM! It works as described.

However, I have made a couple of minor comments.

oikeme committed:
Update details

Added a check to ensure that error details are set if undefined in the API response

oikeme committed:
Update details parameter

Set the default value for the details parameter

yashwin reviewed (approved):
LGTM now. Thanks for making the changes.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7673604

Thank you @oikeme for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7673604).
